### DISCUSSION TITLE
Multi-component solve using Hypre

### DISCUSSION THREAD

**Discussion by ankithadas** [AMReX-Codes/amrex#4273] [General]
Dear AMReX team,

Is it possible to set up a multi-component matrix using the `HypreSolver`? If yes, I am a little bit confused about the stencil size (`MSS`) in that case ? For example, what would the max stencil size be for Navier Stokes Tensor Solve ? 

Kind regards,
Ankith

**Comment by WeiqunZhang** [comment]
It's supposed to work with multiple variables. Suppose there are three variables and 100 points for each variable. The matrix and vectors will have 100*3 rows. `MSS` is the maximum stencil size (i.e., maximum number of non-zeros in a row of the matrix).

Note that `HyperSolve` has not been used that much. If you notice any issues, please let us know.
